src/softmax.jl

```diff
@@ -22,6 +22,20 @@ with additional manipulations enhancing numerical stability.
 For a matrix input `x` it will by default (`dims = 1`) treat it as a batch of vectors,
 with each column independent. Keyword `dims = 2` will instead treat rows independently, and so on.
 
+Since softmax by default acts on a vector instead of a scalar, a syntax different from scalar
+activation functions is to be used when using it in a layer of a neural network. For example,
+a sketch of a convolutional model with a softmax classifier head can be implemented like this:
+
+```jldoctest `
+julia> X = randn(Float32, 4, 4, 3, 9);
+
+julia> model = Chain(Conv((4, 4), 3 => 8), flatten, softmax);
+
+julia> size(model(X))
+(8, 9)
```

Comment:
Also, I think this sketch should go below the basic examples. 

`softmax` accepts any array & can be used without Flux. We should show what it does, and then expand by pointing at how to use it with other things.

Reply:
Also, I'm surprised tests pass with this `jldoctest`, since this does not load Flux. Maybe they aren't being run?

https://github.com/FluxML/NNlib.jl/blob/master/test/runtests.jl#L10-L18